pkg/controller/replicaset/replica_set.go

```diff
@@ -837,15 +837,27 @@ func reportSortingDeletionAgeRatioMetric(filteredPods []*v1.Pod, diff int) {
 // active pods in relatedPods that are colocated on the same node with the pod.
 // relatedPods generally should be a superset of podsToRank.
 func getPodsRankedByRelatedPodsOnSameNode(podsToRank, relatedPods []*v1.Pod) controller.ActivePodsWithRanks {
-	podsOnNode := make(map[string]int)
+	podsOnNodes := make(map[string][]*v1.Pod)
 	for _, pod := range relatedPods {
 		if controller.IsPodActive(pod) {
-			podsOnNode[pod.Spec.NodeName]++
+			podsOnNodes[pod.Spec.NodeName] = append(podsOnNodes[pod.Spec.NodeName], pod)
 		}
 	}
+	for _, podsOnNode := range podsOnNodes {
+		// sort pods on the same node with active and timestamp
+		sort.Sort(controller.ActivePods(podsOnNode))
```

Comment:
This function doesn't sort with logarithmic comparisons

Reply:
In my opinion, there is no need to sort pods on the same node with logarithmic comparisons, which are designed to solve the sort problem across different topologies (such as multiple failure domains).

The pods on different nodes that have the same rank will be sort by logarithmic in the later ActivePodsWithRanks.